mlrun/datastore/targets.py

```diff
@@ -370,6 +372,24 @@ def get_spark_options(self, key_column=None, timestamp_key=None):
     def as_df(self, columns=None, df_module=None):
         raise NotImplementedError()
 
+    def write_dataframe(self, df, key_column=None, timestamp_key=None, **kwargs):
+        if hasattr(df, "rdd"):
+            options = self.get_spark_options(key_column, timestamp_key)
+            options.update(kwargs)
+            df.write.mode("overwrite").save(**options)
+        else:
+            access_key = self._secrets.get(
+                "V3IO_ACCESS_KEY", os.getenv("V3IO_ACCESS_KEY")
+            )
+
+            frames_client = get_frames_client(
+                token=access_key,
+                address=config.v3io_framesd,
+                container=config.model_endpoint_monitoring.container,
```

Comment:
@gtopper why `model_endpoint_monitoring`? the data container is part of the path, i assume u need to use `endpoint, uri = parse_v3io_path(self._target_path)` like above